client/src/Movies/Movie.js

```diff
@@ -25,43 +27,31 @@ export default class Movie extends Component {
         console.error(error);
       });
   };
-  // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
 
-  // saveMovie = () => {
-  //   const addToSavedList = this.props.addToSavedList;
-  //   addToSavedList(this.state.movie)
-  // }
+  //Uncomment this code when you're ready for the stretch problems
+  componentWillReceiveProps(newProps){
```

Comment:
I just found out that `componentWillReceiveProps()` has been deprecated and there is a new pattern for this logic.
more info here >> https://reactjs.org/docs/react-component.html#updating
and here >> https://alligator.io/react/get-derived-state/